src/Illuminate/Foundation/Application.php

```diff
@@ -568,7 +568,15 @@ public function registerConfiguredProviders()
                             return Str::startsWith($provider, 'Illuminate\\');
                         });
 
-        $providers->splice(1, 0, [$this->make(PackageManifest::class)->providers()]);
+        $appProviders = $providers->get(1)
+                            ->partition(function ($provider) {
+                                return Str::startsWith($provider, 'App\\');
```

Comment:
Here u could use `getNamespace` 😃 

```suggestion
                                return Str::startsWith($provider, $this->getNamespace());
```